File: 2350-find-closest-number-to-zero/2350-find-closest-number-to-zero.dart
class Solution {
  int findClosestNumber(List<int> nums) {
    int number =
        nums.map((element) => element.abs()).reduce((a, b) => a < b ? a : b);
    if (!nums.contains(number)) {
      return -number;
    }
    return number;
  }
}

File: 2350-find-closest-number-to-zero/2350-find-closest-number-to-zero.py
class Solution:
    def findClosestNumber(self, nums: list[int]) -> int:
        closet = float('inf')
        for num in nums:
            absNum = abs(num)
            closet = min(absNum, closet)
        if(closet not in nums):
            return -closet
        return closet